### DISCUSSION TITLE
Flask-socketio messages not being received when sent from worker script to web script

### DISCUSSION THREAD

**Discussion by robcowen** [miguelgrinberg/Flask-SocketIO#1974] [Q&A] [answered]
[Duplicate of question posted on StackOverflow earlier today](https://stackoverflow.com/questions/76044575/flask-socketio-messages-not-being-received-when-sent-from-worker-script-to-web-s)

I am using flask-socketio to implement websockets on my Flask app. An intensive function runs on a worker script and is supposed to emit a message to the websocket when it is complete. This message should then be received by the client and used to update the display.

The trouble I'm having is that messages sent from the worker script do not get received by the client. The same messages sent from within the Flask app are received. I assume the problem is that the two different scripts have different websocket connections; how do I make them talk to each other?

**worker.py (running outside of Flask app) - this does not get received by client**

`socketio.emit('edit_plan', {'data': 'testing test'})`

**main.py (running inside Flask app) - this does get received by client**

```
@app.route("/home", methods=['POST', 'GET'])
@login_required
@register_breadcrumb(app, '.', 'Home')
def home():
    # Send websocket test message
    print("Emitting test messages")
    socketio.emit('edit_plan', {'data': 'testing test'})
```

**Front end**

```
componentDidMount() {
    const that = this ;

    // Connect to websocket and await message
    const socket = io();

    socket.on('connect', function() {
      console.log("Connected to websocket")
      console.log(socket.id)
    });

    socket.on('edit_plan', function(msg, cb) {
      console.log(msg)
      alert(msg)
    });
}
```

**Comment by miguelgrinberg** [comment, marked as answer]
Answer on Stack Overflow: https://stackoverflow.com/a/76045858/904393
